service-base/src/test/java/org/eclipse/hono/service/credentials/BaseCredentialsServiceTest.java

```diff
@@ -0,0 +1,187 @@
+        JsonObject testData = new JsonObject();
+        testData.put(TEST_FIELD, "2007-04-05T14:30");
+
+        boolean valid = service.containsValidTimestampIfPresentForField(testData, TEST_FIELD);
+        assertTrue(valid);
+    }
+
+    @Test
+    public void testContainsValidTimestampIfPresentForFieldWhenPresentInvalid() {
+        JsonObject testData = new JsonObject();
+        testData.put(TEST_FIELD, "yakshaver");
+
+        boolean valid = service.containsValidTimestampIfPresentForField(testData, TEST_FIELD);
+        assertFalse(valid);
+    }
+
+    @Test
+    public void testContainsStringValueForFieldWhenNotPresent() {
+        JsonObject testData = new JsonObject();
+
+        boolean valid = service.containsStringValueForField(testData, TEST_FIELD);
+        assertFalse(valid);
+    }
+
+    @Test
+    public void testContainsStringValueForFieldWhenPresent() {
+        JsonObject testData = new JsonObject();
+        testData.put(TEST_FIELD, "yakshaver");
+
+        boolean valid = service.containsStringValueForField(testData, TEST_FIELD);
+        assertTrue(valid);
+    }
+
+    @Test
+    public void testContainsStringValueForFieldWhenPresentEmpty() {
+        JsonObject testData = new JsonObject();
+        testData.put(TEST_FIELD, "");
+
+        boolean valid = service.containsStringValueForField(testData, TEST_FIELD);
+        assertFalse(valid);
+    }
+
+    @Test
+    public void testContainsValidSecretValueWithMissingSecret() {
+        JsonObject testData = new JsonObject();
+
+        boolean valid = service.containsValidSecretValue(testData);
+        assertFalse(valid);
+    }
+
+    @Test
+    public void testContainsValidSecretValueWithNoSecrets() {
+        JsonObject testData = new JsonObject();
+        JsonArray testSecrets = new JsonArray();
+        testData.put(FIELD_SECRETS, testSecrets);
+        
+        boolean valid = service.containsValidSecretValue(testData);
+        assertTrue(valid);
+    }
+
+    @Test
+    public void testContainsValidSecretValueWithEmptySecret() {
+        JsonObject testData = new JsonObject();
+        JsonArray testSecrets = new JsonArray();
+        JsonObject testSecret = new JsonObject();
+        testSecrets.add(testSecret);
+        testData.put(FIELD_SECRETS, testSecrets);
+
+        boolean valid = service.containsValidSecretValue(testData);
+        assertTrue(valid);
+    }
+
+    @Test
+    public void testContainsValidSecretValueWithValidSecret() {
+        JsonObject testData = new JsonObject();
+        JsonArray testSecrets = new JsonArray();
+        JsonObject testSecret = new JsonObject();
+        testSecret.put("key", "myPwd");
+        testSecrets.add(testSecret);
+        testData.put(FIELD_SECRETS, testSecrets);
+
+        boolean valid = service.containsValidSecretValue(testData);
+        assertTrue(valid);
+    }
+
+    private BaseCredentialsService<ServiceConfigProperties> buildBaseCredentialsServiceDummy() {
```

Comment:
Better name this method "createBaseCredentialsService" (skip the dummy), which better complies with the naming used in existing tests already.

Reply:
will be changed